client/components/data/activating-theme/index.js

```diff
@@ -1,49 +1,49 @@
 /**
  * External dependencies
  */
-var React = require( 'react' );
+import React from 'react';
+import { connect } from 'react-redux';
+import omit from 'lodash/object/omit';
 
 /**
  * Internal dependencies
  */
-var CurrentThemeStore = require( 'lib/themes/stores/current-theme' );
-
-function getState( props ) {
-	return {
-		isActivating: CurrentThemeStore.isActivating(),
-		hasActivated: CurrentThemeStore.hasActivated(),
-		currentTheme: CurrentThemeStore.getCurrentTheme( props.siteId )
-	};
-}
+import {
+	isActivating,
+	hasActivated,
+	getCurrentTheme
+} from 'lib/themes/selectors/current-theme';
 
 /**
  * Passes the activating state of themes to the supplied child component.
  */
-var ActivatingThemeData = React.createClass( {
+const ActivatingThemeData = React.createClass( {
 
 	propTypes: {
-		children: React.PropTypes.element.isRequired
-	},
-
-	getInitialState: function() {
-		return getState( this.props );
-	},
-
-	componentWillMount: function() {
-		CurrentThemeStore.on( 'change', this.onActivatingTheme );
-	},
-
-	componentWillUnmount: function() {
-		CurrentThemeStore.off( 'change', this.onActivatingTheme );
-	},
-
-	onActivatingTheme: function() {
-		this.setState( getState( this.props ) );
+		children: React.PropTypes.element.isRequired,
+		// The following props aren't part of this module's public interface,
+		// since they are provided by the `connect()` call that is wrapped around
+		// `ActivatingThemeData` before it is exported.
```

Comment:
This doesn't feel like the way to go; I mean, as we progress, so many components will be `connect`ed — will we paste a three-line comment in each one's `propTypes`?

My first question would be: is this information really necessary? If it is, there's a subsequent question: can we convey that information through code? For instance, some types could be attached to the plain component, and some others to the connected component. If we prefer not to use code, can the comment be reduced to something like `// @connect`, `// @redux`, etc.?